cwms-data-api/src/main/java/cwms/cda/api/StreamController.java

```diff
@@ -0,0 +1,292 @@
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.DSLContext;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+
+import static cwms.cda.data.dao.JooqDao.getDslContext;
+
+public final class StreamController implements CrudHandler {
+
+    static final String TAG = "Streams";
+
+    private final MetricRegistry metrics;
+    private final Histogram requestResultSize;
+
+    public StreamController(MetricRegistry metrics) {
+        this.metrics = metrics;
+        String className = this.getClass().getName();
+        requestResultSize = this.metrics.histogram((name(className, RESULTS, SIZE)));
+    }
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = OFFICE, required = true, description = "Office id for the reservoir project location " +
+                            "associated with the streams."),
+                    @OpenApiParam(name = DIVERTS_FROM_STREAM_ID, description = "Specifies the stream-id of the " +
+                            "stream that the returned streams flow from."),
+                    @OpenApiParam(name = FLOWS_INTO_STREAM_ID, description = "Specifies the stream-id of the " +
+                            "stream that the returned streams flow into."),
+                    @OpenApiParam(name = UNIT_SYSTEM, type = UnitSystem.class, description = UnitSystem.DESCRIPTION)
+            },
+            responses = {
+                    @OpenApiResponse(status = STATUS_200, content = {
+                            @OpenApiContent(isArray = true, type = Formats.JSONV1, from = Stream.class),
+                            @OpenApiContent(isArray = true, type = Formats.JSON, from = Stream.class)
+                    })
+            },
+            description = "Returns matching CWMS Stream Data for a Reservoir Project.",
+            tags = {TAG}
+    )
+    @Override
+    public void getAll(Context ctx) {
+        String office = ctx.queryParam(OFFICE);
+        String divertsFromStream = ctx.queryParam(DIVERTS_FROM_STREAM_ID);
+        String flowsIntoStream = ctx.queryParam(FLOWS_INTO_STREAM_ID);
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            DSLContext dsl = getDslContext(ctx);
+            StreamDao dao = new StreamDao(dsl);
+            String unitSystem = queryParamAsClass(ctx, new String[]{UNIT_SYSTEM, },
+                    String.class, UnitSystem.SI.getValue(), metrics,
+                    name(StreamController.class.getName(), GET_ALL));
+            String stationUnits = "km";
+            if(UnitSystem.EN.getValue().equals(unitSystem)) {
```

Comment:
Why not let the client specify the units rather than parameterizing in the unit system?

Reply:
additionally, CDA defaults to english units, not SI

Reply:
same comments for the getOne

Reply:
NOTE: *always* allow the user to set "specific units" or "the unit system" where units are used. 

Also there is a mechanism in the database to allow the default units for a system to be set and used. We don't fully support that yeah, but it is on the horizon for authorized users so definitely don't hard code unit processing at any point.

Reply:
Updated to use individual unit parameters, rather than unit system. I did notice that I got back an exception if no station units were provided upon retrieval. I updated to default the units to english if none are provided